__docs__/wonder-blocks-dropdown/multi-select.stories.tsx

```diff
@@ -650,3 +655,51 @@ export const CustomOptionItems: StoryComponentType = {
         ),
     ],
 };
+
+/**
+ * This example illustrates how a JSX Element can appear as the label when only
+ * one option is selected and `labelAsText` is undefined.
+ * **Note** this is only supported for SingleSelect and MultiSelect, not Combobox.
+ */
+export const CustomOptionItemsWithNodeLabel: StoryComponentType = {
+    render: function Render() {
+        const [opened, setOpened] = React.useState(true);
+        const [selectedValues, setSelectedValues] = React.useState<
+            Array<string>
+        >([]);
+
+        const handleChange = (selectedValues: Array<string>) => {
+            setSelectedValues(selectedValues);
+        };
+
+        const handleToggle = (opened: boolean) => {
+            setOpened(opened);
+        };
+
+        return (
+            <MultiSelect
```

Comment:
I played around with the changes and found if we set `isFilterable` in this story, the filtering functionality doesn't work as expected! However, if we add a `labelAsText` prop to the `OptionItem` components with the content (ie. the `locale` in this example), filtering works as expected. 

I'm wondering, should we use `labelAsText` prop when we use a node label? And if so, would that solve for the original issue where the select opener is labeled with an empty string? Let me know though if there's some context I'm missing!

cc: @jandrade , I think he was running into similar things when he worked on the combobox!

Reply:
That's a good callout, and the proposed solution does work solve the original issue.

Reply:
@beaesguerra Actually I don't think this works. If you add `labelAsText` to the option item, you see plain text as the label for the menu. 
<img width="591" alt="Screenshot 2024-11-13 at 9 07 33 AM" src="https://github.com/user-attachments/assets/a82a54d1-b127-4150-9389-4e9879044f9a">

That said, my specific use case does not use filtering.

Reply:
I understand that the specific use case in this PR doesn't need filtering, but I worry that adding this feature while introducing a known issue might be counterproductive if there's another call site that needs both things (custom options + filtering).

I'd really recommend applying the fix/change in this PR before landing this change.

I was thinking that it might be solved by changing the filter logic to use `.toString()` after the `getSelectOpenerLabel` invocation. I'm not fully sure if this is the right solution but it might be worth trying it out for both `MultiSelect` and `SingleSelect`.